### PR TITLE
Added files from openshift-knative/must-gather to serverless-operator

### PR DESCRIPTION
**Pull request by Kaustubh-pande**
Fixes JIRA # [SRVCOM-3320](https://issues.redhat.com/browse/SRVCOM-3320)

more info found in https://redhat-internal.slack.com/archives/CKR568L8G/p1725354894792959

### CODE REVIEW

File: must-gather/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM quay.io/openshift/origin-must-gather:4.11.0
```
Comment (pierDipi) [inline must-gather/Dockerfile:1]: Can we use the production base image?

Comment (pierDipi) [reply must-gather/Dockerfile:1]: also 4.11 is very old

**Comment by pierDipi** [comment]
@Kaustubh-pande any update on this?

**Comment by Kaustubh-pande** [comment]
adding must-gather to hack repo https://github.com/openshift-knative/hack/pull/370

File: must-gather/Dockerfile (hunk 1 above)
Comment (Kaustubh-pande) [reply must-gather/Dockerfile:1]: Latest Dockerfile template will be in hack repo

**Comment by Kaustubh-pande** [comment]
Validate / Generated files are committed (pull_request) Test will depend on https://github.com/openshift-knative/hack/pull/370

File: must-gather/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ARG RUNTIME=registry.access.redhat.com/ubi8/ubi-minimal
+FROM $MUSTGATHER AS builder
+
+FROM $RUNTIME
+
+COPY --from=builder /usr/bin/oc /usr/bin/oc
+COPY --from=builder /usr/bin/gather /usr/bin/gather_original
+
+# Copy all collection scripts to /usr/bin
+COPY bin/* /usr/bin/
```
Comment (mgencur) [inline must-gather/Dockerfile:12]:
```suggestion
COPY must-gather/bin/* /usr/bin/
```

File: Makefile [hunk 3]
```diff
@@ -340,6 +340,14 @@ generate-dockerfiles: install-tool-generate
...
 	git apply knative-operator/dockerfile.patch
 	git apply openshift-knative-operator/dockerfile.patch
+	generate-must-gather-dockerfile
+generate-must-gather-dockerfile: install-tool-generate
+	$(shell go env GOPATH)/bin/generate \
+		--generators must-gather-dockerfile \
+		--project-file olm-catalog/serverless-operator/project.yaml \
+		--includes must-gather \
+		--output /tmp/serverless-operator-generator/
+	cp  /tmp/serverless-operator-generator/ci-operator/knative-images/must-gather/Dockerfile must-gather/Dockerfile
```
Comment (creydr) [inline Makefile:350]: I am getting 
```
$ make generate-must-gather-dockerfile
GOFLAGS='' go install github.com/openshift-knative/hack/cmd/generate@latest
.../generate \
        --generators must-gather-dockerfile \
        --project-file olm-catalog/serverless-operator/project.yaml \
        --includes must-gather \
        --output /tmp/serverless-operator-generator/
cp  /tmp/serverless-operator-generator/ci-operator/knative-images/must-gather/Dockerfile must-gather/Dockerfile
cp: cannot stat '/tmp/serverless-operator-generator/ci-operator/knative-images/must-gather/Dockerfile': No such file or directory
```

When I create `/tmp/serverless-operator-generator` upfront it works

File: Makefile [hunk 4]
```diff
@@ -340,6 +340,14 @@ generate-dockerfiles: install-tool-generate
 
 	git apply knative-operator/dockerfile.patch
 	git apply openshift-knative-operator/dockerfile.patch
+	generate-must-gather-dockerfile
+generate-must-gather-dockerfile: install-tool-generate
```
Comment (pierDipi) [inline Makefile:344]:
```suggestion
```

File: Makefile (hunk 3 above)
Comment (pierDipi) [reply Makefile]: It should work with the new commit

**Comment by pierDipi** [comment]
It works now, let's merge

**Pull request merged**